### ISSUE TITLE
question about the hyper-parameters

### ISSUE THREAD

**Issue by onlytailei** [ikostrikov/pytorch-a3c#27]
In [train.py](https://github.com/ikostrikov/pytorch-a3c/blob/master/train.py)
Line 95 and Line 107 both have a 0.5 decay parameter for value loss.
That's why? The original paper does not describe that.

**Comment by Kaixhin** [comment]
One is actually a hyperparameter that [the authors failed to mention in the paper](https://github.com/muupan/async-rl/wiki). Not exactly sure why it appears a second time though (I've opted to use it only once in my own code, on the loss itself), so I'd like to know too.

**Comment by ikostrikov** [comment]
It's a typo. I will fix it.

**Comment by onlytailei** [comment]
And do you think the GAE implementation is correct?
It seems that you only use A(n).
But A(GAE) = A(1)+A(2)+A(3)+.....A(n).

https://arxiv.org/pdf/1506.02438.pdf

**Comment by dgriff777** [comment]
value_loss = value_loss + 0.5 * advantage.pow(2)

this is huber loss

**Comment by Kaixhin** [comment]
@dgriff777 It should be just a weighted MSE loss (as in the paper), not the Huber loss (which was used for the DQN, but not A3C). You can see the code can be written as `L += 1/2 * (R - V)^2`:

```py
advantage = R - values[i]
value_loss = value_loss + 0.5 * advantage.pow(2)
```

**Comment by dgriff777** [comment]
Yeah my mistake was testing both out. L2 it is 👍They the same thing most the time lol. Yeah paper does Mse but always good to mix it up and do better than the papers😉
